### PR TITLE
Use go modules and Go 1.15

### PR DESCRIPTION
**Pull request by mythri-garaga**
### Summary

This PR carries #2493 
 - moves agent agent away from godep for dependency management and using go modules 
### Implementation details

- from agent/
```
go mod init
go mod tidy
rm -rf vendor
go mod vendor
```
- updated  updated .travis.yml and Makefile to use -gcflags=all=-d=checkptr=0 flag  while running unit test and integ tests
- updated go version to 1.15 in travis
- In go.mod, golang.org/x/sys use golang.org/x/sys v0.0.0-20190830141801-acfa387b8d69 as the API was breaking
- deleted agent/Gopkg.lock  and agent/Gopkg.toml
- refactored agent/api/task/util_test.go to avoid string conversion error with go 1.15
- updated Dockerfiles to build agent using Go 1.15
- migrating tool dependencies to be vendored by go modules 

### Testing

- `make test` passes locally
- windows integ tests are expected to fail since ttb windows userdata has to updated to use Go 1.15

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mythri-garaga** [comment]
Windows functional tests failing with error from transcript log:

```
PS>TerminatingError(Write-S3Object): "The file indicated by the FilePath property does not exist!"
Write-S3Object : The file indicated by the FilePath property does not exist!
At C:\Windows\TEMP\UserScript.ps1:107 char:3
+   Write-S3Object -Region "${REGION}" -BucketName "${S3BUCKET}" -Key " ...
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Amazon.PowerShe...eS3ObjectCmdlet:WriteS3ObjectCmdlet) 
[Write-S3Object], InvalidOperationException
    + FullyQualifiedErrorId : System.ArgumentException,Amazon.PowerShell.Cmdlets.S3.WriteS3ObjectCmdlet
Write-S3Object : The file indicated by the FilePath property does not exist!
At C:\Windows\TEMP\UserScript.ps1:107 char:3
+   Write-S3Object -Region "${REGION}" -BucketName "${S3BUCKET}" -Key " ...
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Amazon.PowerShe...eS3ObjectCmdlet:WriteS3ObjectCmdlet) [Write-S3Objec
   t], InvalidOperationException
    + FullyQualifiedErrorId : System.ArgumentException,Amazon.PowerShell.Cmdlets.S3.WriteS3ObjectCmdlet

**Review by sharanyad — APPROVED** [review]
:shipit:

File: scripts/dockerfiles/Dockerfile.cleanbuild [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-FROM golang:1.12
+FROM golang:1.12.4
```
Comment (sparrc) [inline scripts/dockerfiles/Dockerfile.cleanbuild:14]: should this be `FROM golang:1.15` ?

Comment (mythri-garaga) [reply scripts/dockerfiles/Dockerfile.cleanbuild:14]: It should be, I will be sending a follow up CR for all the dockerfile updates

**Comment by sparrc** [comment]
It looks like all of our Dockerfiles in scripts/dockerfiles are still on 1.12, which means our release would end up being built with 1.12.

Was the intention to upgrade building our releases with Golang 1.15 as part of a separate PR?

**Comment by sparrc** [comment]
We can delete any references to the github.com/golang/dep too, such as here: https://github.com/aws/amazon-ecs-agent/blob/master/Makefile#L314-L321

**Comment by mythri-garaga** [comment]
> It looks like all of our Dockerfiles in scripts/dockerfiles are still on 1.12, which means our release would end up being built with 1.12.
> 
> Was the intention to upgrade building our releases with Golang 1.15 as part of a separate PR?

Yes, changes to Dockerfiles are going to be in a seperate PR.

**Comment by mythri-garaga** [comment]
> We can delete any references to the github.com/golang/dep too, such as here: https://github.com/aws/amazon-ecs-agent/blob/master/Makefile#L314-L321

Thanks for pointing this out. Updated!

**Pull request closed without merging**